src/poprox_recommender/evaluation/generate.py

```diff
@@ -0,0 +1,139 @@
+"""
+Generate recommendations for offline test data.
+
+Usage:
+    poprox_recommender.evaluation.generate [options]
+
+Options:
+    -v, --verbose
+            enable verbose diagnostic logs
+    --log-file=FILE
+            write log messages to FILE
+    -o FILE, --output=FILE
+            write output to FILE [default: outputs/recommendations.parquet]
+"""
+
+# pyright: basic
+from __future__ import annotations
+
+import logging
+import logging.config
+
+import pandas as pd
+from docopt import docopt
+from tqdm import tqdm
+
+from poprox_concepts.api.recommendations import RecommendationRequest
+from poprox_concepts.domain import ArticleSet
+from poprox_recommender.data.mind import TEST_REC_COUNT, MindData
+from poprox_recommender.default import fallback_pipeline, personalized_pipeline
+from poprox_recommender.logging_config import setup_logging
+from poprox_recommender.pipeline import PipelineState, RecommendationPipeline
+
+logger = logging.getLogger("poprox_recommender.test_offline")
+
+# long-term TODO:
+# - support other MIND data (test?)
+# - support our data
+
+
+def extract_recs(
```

Comment:
This function is no longer going to work with the new pipeline & pipeline state, so we need to update it to that.

Reply:
@mdekstrand Since you know the new pipeline abstraction best, maybe you could add a commit here to make that adjustment?